### DISCUSSION TITLE
Integrating Fore with Vue

### DISCUSSION THREAD

**Discussion by tomaszmac** [Jinntec/Fore#103] [Q&A]
Hello,

I'm creating app with Vue and I'm currently trying to include that lib in my project. 

I tired to include it in my main.js as 
```
import Fore from '@jinntec/fore';
Vue.use(Fore);
```

But it throws an error in a console:
```
Uncaught TypeError: Cannot read properties of undefined (reading 'install')
    at Function.Vue.use (vue.esm.js?4b29:5129:1)
    at eval (main.js?b0ca:11:1)
    at Module../src/main.js (app.js:1227:1)
    at __webpack_require__ (app.js:4415:33)
    at fn (app.js:4667:21)
```
So I tired adding it to index.html with cdn option. It kinda works, but only if I use v-html on a div for example. Problem is that it doesn't have any styles. When I use it in _template_ it stops working. There is something showing up but buttons are just creating errors. This is the copy of: https://jinntec.github.io/Fore/demo/todo.html.
![image](https://user-images.githubusercontent.com/64752217/154983769-831b73e6-5957-491d-bf8a-7eb9ae192590.png).
Throwing errors:
```
1. [Vue warn]: Unknown custom element: <fx-repeat> - did you register the component correctly? For recursive components, make sure to provide the "name" option.
2. Uncaught (in promise) TypeError: Failed to execute 'evaluateXPath': xpathExpression must be a string.
3. Uncaught DOMException: Failed to construct 'CustomElement': The result must not have attributes
```
I also want to add Vuetify elements into Fore like v-btn czy v-text-input so I can't use the v-html.

**Comment by JoernT** [comment]
to be honest i can't help a lot with Vue cause i have no experience with it.

However if you share some code i'll have a look at it. 

Further i guess it might be helpful to understand your motivation. Why do you want to integrate them? What are you missing from Vue that Fore can give you (and vice versa)? These would be interesting points for me. So, do you want Fore to be part of your Vue app or use Vue components as Fore controls?

Thanks,

Joern
